be/src/runtime/snapshot_loader.cpp

```diff
@@ -655,7 +655,13 @@ Status SnapshotLoader::move(const std::string& snapshot_path, const TabletShared
         // link files one by one
         // files in snapshot dir will be moved in snapshot clean process
         std::vector<std::string> linked_files;
+        std::string dcg_file;
         for (auto& file : snapshot_files) {
+            // A tablet can have at most one dcgs snapshot file
+            if (_end_with(file, ".dcgs_snapshot")) {
```

Comment:
1. If the schema change failed when to clear the `.dcgs_snapshot`
2. If the schema change failed and the BE restarted, when to clear the `.dcgs_snapshot`?

Reply:
Here is the restore related code. .dcgs_snapshot is the file contain the metadata of the backup table's dcg information. The . dcgs_snapshot file is always in snapshot path, which will be remove later in the restore procedure.